src/api/nft-namespace.ts

```diff
@@ -0,0 +1,646 @@
+import { BigNumber, BigNumberish } from '@ethersproject/bignumber';
+import {
+  GetBaseNftsForContractOptions,
+  GetBaseNftsForOwnerOptions,
+  GetFloorPriceResponse,
+  GetNftsForContractOptions,
+  GetNftsForOwnerOptions,
+  GetOwnersForContractResponse,
+  GetOwnersForNftResponse,
+  NftContractBaseNftsResponse,
+  NftContractNftsResponse,
+  NftTokenType,
+  OwnedBaseNft,
+  OwnedBaseNftsResponse,
+  OwnedNft,
+  OwnedNftsResponse,
+  RefreshContractResult,
+  RefreshState
+} from '../types/types';
+import { BaseNft, Nft, NftContract } from './nft';
+import { AlchemyConfig } from './alchemy-config';
+import { paginateEndpoint, requestHttpWithBackoff } from '../internal/dispatch';
+import {
+  RawBaseNft,
+  RawGetBaseNftsForNftContractResponse,
+  RawGetBaseNftsResponse,
+  RawGetNftsForNftContractResponse,
+  RawGetNftsResponse,
+  RawGetOwnersForNftContractResponse,
+  RawNft,
+  RawNftContract,
+  RawNftContractBaseNft,
+  RawNftContractNft,
+  RawOwnedBaseNft,
+  RawOwnedNft,
+  RawReingestContractResponse
+} from '../internal/raw-interfaces';
+import { AlchemyApiType } from '../util/const';
+import {
+  getBaseNftFromRaw,
+  getNftContractFromRaw,
+  getNftFromRaw
+} from '../util/util';
+
+export class NftNamespace {
+  constructor(private readonly config: AlchemyConfig) {}
+
+  /**
+   * Get the NFT metadata associated with the provided parameters.
+   *
+   * @param contractAddress - The contract address of the NFT.
+   * @param tokenId - Token id of the NFT.
+   * @param tokenType - Optionally specify the type of token to speed up the query.
+   * @public
+   */
+  async getNftMetadata(
+    contractAddress: string,
+    tokenId: BigNumberish,
+    tokenType?: NftTokenType
+  ): Promise<Nft> {
+    const response = await requestHttpWithBackoff<GetNftMetadataParams, RawNft>(
+      this.config,
+      AlchemyApiType.NFT,
+      'getNFTMetadata',
+      {
+        contractAddress,
+        tokenId: BigNumber.from(tokenId!).toString(),
+        tokenType: tokenType !== NftTokenType.UNKNOWN ? tokenType : undefined
+      }
+    );
+    return getNftFromRaw(response, contractAddress);
+  }
+
+  /**
+   * Get the NFT collection metadata associated with the provided parameters.
+   *
+   * @param contractAddress - The contract address of the NFT.
+   * @public
+   */
+  async getNftContractMetadata(contractAddress: string): Promise<NftContract> {
```

Comment:
This should be getContractMetadata post-namespace!

Reply:
done